src/test/java/com/fasterxml/jackson/databind/ObjectReaderTest.java

```diff
@@ -805,4 +808,111 @@ public int hashCode() {
             return _delegate.hashCode();
         }
     }
+
+
+    public void testReadValueFromFile() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().readerFor(FilePerson.class).readValue(file);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFile2() throws IOException {
+        File file = _createFileWithNameAndJson(
+            "testReadValueFromFile2.json",
+            a2q("{ 'name': 'John Doe'}"));
+
+        FilePerson bean = newJsonMapper().reader().readValue(file, FilePerson.class);
+
+        assertEquals("John Doe", bean.name);
+        _cleanUpFromFileSystem(file);
+    }
+
+    public void testReadValueFromFileWithFormatDetection() throws IOException {
```

Comment:
Let's leave out format-detection cases since that functionaltity is dropped from 3.0.